components/docker-auth/cmd/authz/authorization.go

```diff
@@ -19,88 +19,61 @@
+	"go.uber.org/zap"
 
-const (
-	logFile = "/extension-logs/authz-ext.log"
+	"github.com/cellery-io/cellery-hub/components/docker-auth/pkg/auth"
+	"github.com/cellery-io/cellery-hub/components/docker-auth/pkg/db"
+	"github.com/cellery-io/cellery-hub/components/docker-auth/pkg/extension"
 )
 
-func main() {
-	err := os.MkdirAll("/extension-logs", os.ModePerm)
-	file, err := os.OpenFile(logFile, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
-	defer func() {
-		err = file.Close()
-		if err != nil {
-			log.Printf("Error while closing the file : %s\n", err)
-			os.Exit(2)
-		}
-	}()
-	if err != nil {
-		log.Println("Error creating the file :", err)
-		os.Exit(1)
-	}
-	log.SetOutput(file)
-
-	execId, err := extension.GetExecID()
-	if err != nil {
-		log.Printf("Error in generating the execId : %s\n", err)
-		os.Exit(extension.ErrorExitCode)
-	}
-
-	accessToken := extension.ReadStdIn()
-	log.Printf("[%s] Access token received\n", execId)
+var authzLogger *zap.SugaredLogger
 
-	url := resolveAuthorizationUrl(execId)
-	if url == "" {
-		log.Printf("[%s] Authorization end point not found. Exiting with error exit code", execId)
-		os.Exit(extension.ErrorExitCode)
-	}
-
-	payload := strings.NewReader(accessToken)
-	log.Printf("[%s] Calling %s with accessToken : %s as payload", execId, url, accessToken)
+type PluginAuthz struct {
+}
 
-	req, _ := http.NewRequest("POST", url, payload)
-	req.Header.Add(extension.ExecIdHeaderName, execId)
+func (c *PluginAuthz) Stop() {
+}
 
-	res, _ := http.DefaultClient.Do(req)
+func (c *PluginAuthz) Name() string {
+	return "plugin authz"
+}
 
-	defer func() {
-		err := res.Body.Close()
+func (c *PluginAuthz) Authorize(ai *api.AuthRequestInfo) ([]string, error) {
+	if authzLogger == nil {
+		var err error
+		authzLogger, err = extension.NewLogger()
 		if err != nil {
-			log.Printf("[%s] Error occured while closing the response received from auth server "+
-				" : %v\n", execId, err)
+			log.Fatalf("can't initialize zap logger: %v", err)
 		}
-	}()
+	}
+	return doAuthorize(ai, authzLogger)
+}
 
-	log.Printf("[%s] Response received from the auth server with the status code : %d", execId, res.StatusCode)
+var Authz PluginAuthz
 
-	if res.StatusCode == http.StatusUnauthorized {
-		log.Printf("[%s] Unauthorized request. Exiting with error exit code", execId)
-		os.Exit(extension.ErrorExitCode)
+func doAuthorize(ai *api.AuthRequestInfo, logger *zap.SugaredLogger) ([]string, error) {
+	execId, err := extension.GetExecID(logger)
+	if err != nil {
+		return nil, fmt.Errorf("error in generating the execId : %s", err)
 	}
-	if res.StatusCode == http.StatusOK {
-		log.Printf("[%s] Authorized request. Exiting with success exit code", execId)
-		os.Exit(extension.SuccessExitCode)
+	logger.Debugf("Authorization logic reached. User will be authorized")
+	dbConnectionPool, err := db.GetDbConnectionPool(logger)
+	if err != nil {
+		return nil, fmt.Errorf("error while establishing database connection pool")
```

Comment:
Add the original err when returning new error with messages

Reply:
Fixed